lib/include/net/cne_ether.h

```diff
@@ -0,0 +1,515 @@
+
+/**
+ * Ethernet VLAN Header.
+ * Contains the 16-bit VLAN Tag Control Identifier and the Ethernet type
+ * of the encapsulated frame.
+ */
+struct cne_vlan_hdr {
+    uint16_t vlan_tci;  /**< Priority (3) + CFI (1) + Identifier Code (12) */
+    uint16_t eth_proto; /**< Ethernet type of encapsulated frame. */
+} __cne_packed;
+
+/* Ethernet frame types */
+#define CNE_ETHER_TYPE_IPV4            0x0800 /**< IPv4 Protocol. */
+#define CNE_ETHER_TYPE_IPV6            0x86DD /**< IPv6 Protocol. */
+#define CNE_ETHER_TYPE_ARP             0x0806 /**< Arp Protocol. */
+#define CNE_ETHER_TYPE_RARP            0x8035 /**< Reverse Arp Protocol. */
+#define CNE_ETHER_TYPE_VLAN            0x8100 /**< IEEE 802.1Q VLAN tagging. */
+#define CNE_ETHER_TYPE_QINQ            0x88A8 /**< IEEE 802.1ad QinQ tagging. */
+#define CNE_ETHER_TYPE_PPPOE_DISCOVERY 0x8863 /**< PPPoE Discovery Stage. */
+#define CNE_ETHER_TYPE_PPPOE_SESSION   0x8864 /**< PPPoE Session Stage. */
+#define CNE_ETHER_TYPE_ETAG            0x893F /**< IEEE 802.1BR E-Tag. */
+#define CNE_ETHER_TYPE_1588            0x88F7
+/**< IEEE 802.1AS 1588 Precise Time Protocol. */
+#define CNE_ETHER_TYPE_SLOW  0x8809 /**< Slow protocols (LACP and Marker). */
+#define CNE_ETHER_TYPE_TEB   0x6558 /**< Transparent Ethernet Bridging. */
+#define CNE_ETHER_TYPE_LLDP  0x88CC /**< LLDP Protocol. */
+#define CNE_ETHER_TYPE_MPLS  0x8847 /**< MPLS ethertype. */
+#define CNE_ETHER_TYPE_MPLSM 0x8848 /**< MPLS multicast ethertype. */
+
+/* Ethernet frame types */
+#define __ETHER_TYPE_IPV4            0x0008 /**< IPv4 Protocol. */
+#define __ETHER_TYPE_IPV6            0xDD86 /**< IPv6 Protocol. */
+#define __ETHER_TYPE_ARP             0x0608 /**< Arp Protocol. */
+#define __ETHER_TYPE_RARP            0x3580 /**< Reverse Arp Protocol. */
+#define __ETHER_TYPE_VLAN            0x0081 /**< IEEE 802.1Q VLAN tagging. */
+#define __ETHER_TYPE_QINQ            0xA888 /**< IEEE 802.1ad QinQ tagging. */
+#define __ETHER_TYPE_PPPOE_DISCOVERY 0x6388 /**< PPPoE Discovery Stage. */
+#define __ETHER_TYPE_PPPOE_SESSION   0x6488 /**< PPPoE Session Stage. */
+#define __ETHER_TYPE_ETAG            0x3F89 /**< IEEE 802.1BR E-Tag. */
+#define __ETHER_TYPE_1588            0xF788
+/**< IEEE 802.1AS 1588 Precise Time Protocol. */
+#define __ETHER_TYPE_SLOW  0x0988 /**< Slow protocols (LACP and Marker). */
+#define __ETHER_TYPE_TEB   0x5865 /**< Transparent Ethernet Bridging. */
+#define __ETHER_TYPE_LLDP  0xCC88 /**< LLDP Protocol. */
+#define __ETHER_TYPE_MPLS  0x4788 /**< MPLS ethertype. */
+#define __ETHER_TYPE_MPLSM 0x4888 /**< MPLS multicast ethertype. */
+
+/**
+ * Check if two Ethernet addresses are the same.
+ *
+ * @param ea1
+ *  A pointer to the first ether_addr structure containing
+ *  the ethernet address.
+ * @param ea2
+ *  A pointer to the second ether_addr structure containing
+ *  the ethernet address.
+ *
+ * @return
+ *  True  (1) if the given two ethernet address are the same;
+ *  False (0) otherwise.
+ */
+static inline int
```

Comment:
Are the functions in this file intended to be static inline or 
```suggestion
 CNDP_API __cne_always_inline  int
```

Reply:
Convert them to suggested code.